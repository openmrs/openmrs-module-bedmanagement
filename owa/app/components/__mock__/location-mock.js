Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});