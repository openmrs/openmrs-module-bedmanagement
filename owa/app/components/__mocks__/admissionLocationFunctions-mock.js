import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
