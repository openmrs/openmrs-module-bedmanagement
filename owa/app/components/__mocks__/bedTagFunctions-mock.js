import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
