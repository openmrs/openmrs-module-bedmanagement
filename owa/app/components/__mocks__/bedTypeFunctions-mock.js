import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
