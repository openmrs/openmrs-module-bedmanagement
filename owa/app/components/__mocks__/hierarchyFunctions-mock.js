import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;