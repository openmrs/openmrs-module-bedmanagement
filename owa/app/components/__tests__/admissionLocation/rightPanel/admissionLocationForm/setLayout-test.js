import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import admissionLocationFunctionsMock from 'components/__mocks__/admissionLocationFunctions-mock';
import SetBedLayout from '../../../../admissionLocation/rightPanel/admissionLocationForm/setBedLayout';

require('components/__mocks__/location-mock');
const testProps = {
    admissionLocationFunctions: admissionLocationFunctionsMock
};

describe('SetBedLayout', () => {
    beforeAll(() => {
        var mock = new MockAdapter(axios);
        const data = {
            'ward': {
                'uuid': 'e48fb2b3-d490-11e5-b193-0800270d80ce',
                'display': 'General Ward - Room 2',
                'name': 'New',
                'description': 'General Ward room number 2',
                'tags': [{
                    'uuid': 'a675e840-d225-11e4-9c67-080027b662ec',
                    'display': 'Admission Location',
                    'links': [{
                        'rel': 'self',
                        'uri': 'http://192.168.33.10/openmrs/ws/rest/v1/locationtag/a675e840-d225-11e4-9c67-080027b662ec'
                    }]
                }],
                'parentLocation': {
                    'uuid': 'baf7bd38-d225-11e4-9c67-080027b662ec',
                    'display': 'General Ward',
                    'links': [{
                        'rel': 'self',
                        'uri': 'http://192.168.33.10/openmrs/ws/rest/v1/location/baf7bd38-d225-11e4-9c67-080027b662ec'
                    }]
                },
                'childLocations': [],
                'retired': false,
            },
            'bedLocationMappings': [{
                'rowNumber': 1,
                'columnNumber': 1,
                'bedNumber': null,
                'bedId': null,
                'bedUuid': null,
                'status': null,
                'bedType': null
            }, {
                'rowNumber': 1,
                'columnNumber': 2,
                'bedNumber': null,
                'bedId': null,
                'bedUuid': null,
                'status': null,
                'bedType': null
            }, {
                'rowNumber': 1,
                'columnNumber': 3,
                'bedNumber': null,
                'bedId': null,
                'bedUuid': null,
                'status': null,
                'bedType': null
            }]
        };

        mock.onPost('https://192.168.33.10/openmrs/ws/rest/v1/admissionLocation/baf7bd38-d225-11e4-9c67-080027b662ec')
            .reply(200, data);
    });

    it('Should display set layout form properly', () => {
        const setLayoutForm = shallow(<SetBedLayout activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec'
            admissionLocationFunctions={testProps.admissionLocationFunctions}/>);

        expect(shallowToJson(setLayoutForm)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnCancelEventHandler = jest.spyOn(SetBedLayout.prototype, 'cancelEventHandler');
        const spyOnSubmitHandler = jest.spyOn(SetBedLayout.prototype, 'onSubmitHandler');
        const spyOnChangeRowField = jest.spyOn(SetBedLayout.prototype, 'onChangeRowField');
        const spyOnChangeColumnField = jest.spyOn(SetBedLayout.prototype, 'onChangeColumnField');

        const spySetState = jest.spyOn(testProps.admissionLocationFunctions, 'setState');
        const setLayoutForm = mount(<SetBedLayout activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec'
            admissionLocationFunctions={testProps.admissionLocationFunctions}/>);

        setLayoutForm.find('#row-field').simulate('change');
        expect(spyOnChangeRowField).toHaveBeenCalled();
        setLayoutForm.find('#column-field').simulate('change');
        expect(spyOnChangeColumnField).toHaveBeenCalled();

        setLayoutForm.find('.btn-danger').simulate('click');
        expect(spyOnCancelEventHandler).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({
            activePage: 'listing',
            pageData: {},
            activeUuid: 'baf7bd38-d225-11e4-9c67-080027b662ec'
        });

        setLayoutForm.find('input[type=\'submit\']').simulate('submit');
        expect(spyOnSubmitHandler).toHaveBeenCalled();
    });
});