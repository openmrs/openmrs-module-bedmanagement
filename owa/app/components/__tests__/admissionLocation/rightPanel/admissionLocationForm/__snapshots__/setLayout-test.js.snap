// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetBedLayout Should display set layout form properly 1`] = `
<div
  className="main-container"
>
  <fieldset
    className="admission-location-form"
  >
    <legend>
        Set Layout  
    </legend>
    <div
      className="block-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-block"
        >
          <label
            className="form-title inline"
          >
            Location:
          </label>
          <span>
            General Ward
          </span>
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Rows:
          </label>
          <input
            id="row-field"
            onChange={[Function]}
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Column:
          </label>
          <input
            id="column-field"
            onChange={[Function]}
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-block"
        >
          <input
            className="form-btn float-left margin-right"
            disabled={false}
            name="submit"
            type="submit"
            value="Save"
          />
          <input
            className="form-btn float-left"
            name="cancel"
            onClick={[Function]}
            type="button"
            value="Cancel"
          />
        </div>
      </form>
    </div>
  </fieldset>
</div>
`;
